activerecord/test/cases/calculations_test.rb

```diff
@@ -248,6 +248,10 @@ def test_count_with_eager_loading_and_custom_order_and_distinct
     assert_queries(1) { assert_equal 11, posts.count(:all) }
   end
 
+  def test_count_all_with_distinct_selected_value_and_join
+    assert_equal 5, Account.select("DISTINCT accounts.id").joins(:firm).count(:all)
```

Comment:
Just out of curiosity: would `Account.distinct.joins(:firm).count` also return 5?

Reply:
Thanks for this great feedback. I'll double check this and address your other comment above.